src/test/java/edu/hm/hafner/analysis/ModuleDetectorTest.java

```diff
@@ -124,14 +129,186 @@ void shouldIdentifyModuleByReadingMavenPomWithoutName() {
                 .isEqualTo(StringUtils.EMPTY);
     }
 
+    @Test
+    void shouldIdentifyModuleByReadingGradlePath() {
+        FileSystem factory = createFileSystemStub(stub -> {
+            when(stub.find(any(), anyString())).thenReturn(
+                    new String[] {PATH_PREFIX_GRADLE + ModuleDetector.BUILD_GRADLE});
+        });
+
+        ModuleDetector detector = new ModuleDetector(ROOT, factory);
+
+        assertThat(detector.guessModuleName(PREFIX + (PATH_PREFIX_GRADLE + "build/reports/something.txt")))
+                .isEqualTo(EXPECTED_GRADLE_MODULE_ROOT_BY_PATH);
+        assertThat(detector.guessModuleName(PREFIX + "build/reports/something.txt"))
+                .isEqualTo(StringUtils.EMPTY);
+    }
+
+    @Test
+    void shouldIdentifyModuleByFindingClosestGradlePath() {
+        FileSystem factory = createFileSystemStub(stub -> {
+            when(stub.find(any(), anyString())).thenReturn(new String[] {
+                    PATH_PREFIX_GRADLE + ModuleDetector.BUILD_GRADLE,
+                    PATH_PREFIX_GRADLE + "moduleB/" + ModuleDetector.BUILD_GRADLE,
+                    PATH_PREFIX_GRADLE + "a-module/" + ModuleDetector.BUILD_GRADLE,
+            });
+        });
+
+        ModuleDetector detector = new ModuleDetector(ROOT, factory);
+        String gradleWorkspace = PREFIX + PATH_PREFIX_GRADLE;
+
+        assertThat(detector.guessModuleName(
+                gradleWorkspace + "a-module/build/reports/something.txt"))
+                .isEqualTo(EXPECTED_GRADLE_MODULE_A);
+        assertThat(detector.guessModuleName(
+                gradleWorkspace + "moduleB/build/reports/something.txt"))
+                .isEqualTo(EXPECTED_GRADLE_MODULE_B);
+        assertThat(detector.guessModuleName(gradleWorkspace + "build/reports/something.txt"))
+                .isEqualTo(EXPECTED_GRADLE_MODULE_ROOT_BY_PATH);
+        assertThat(detector.guessModuleName(PREFIX + "build/reports/something.txt"))
+                .isEqualTo(StringUtils.EMPTY);
+    }
+
+    @Test
+    void shouldIdentifyModuleByReadingGradleKtsPath() {
+        FileSystem factory = createFileSystemStub(stub -> {
+            when(stub.find(any(), anyString())).thenReturn(
+                    new String[] {PATH_PREFIX_GRADLE + ModuleDetector.BUILD_GRADLE_KTS});
+        });
+
+        ModuleDetector detector = new ModuleDetector(ROOT, factory);
+
+        assertThat(detector.guessModuleName(PREFIX + (PATH_PREFIX_GRADLE + "build/reports/something.txt")))
+                .isEqualTo(EXPECTED_GRADLE_MODULE_ROOT_BY_PATH);
+        assertThat(detector.guessModuleName(PREFIX + "build/reports/something.txt"))
+                .isEqualTo(StringUtils.EMPTY);
+    }
+
+    @Test
+    void shouldIdentifyModuleByReadingGradleSettings() {
+        FileSystem factory = createFileSystemStub(stub -> {
+            when(stub.find(any(), anyString())).thenReturn(new String[] {
+                    PATH_PREFIX_GRADLE + ModuleDetector.BUILD_GRADLE,
```

Comment:
I should probably remove this, since it is tested in the next test, ensuring that settings.gradle takes precedence over the build.gradle base-directory method.  But 99% of all projects will have both build.gradle and settings.gradle.